x/mint/keeper/keeper_test.go

```diff
@@ -60,109 +60,131 @@ func TestKeeperTestSuite(t *testing.T) {
 		},
 	}
-	suite.App.MintKeeper.SetParams(suite.Ctx, params)
-
-	// Create record
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10000)}
-	suite.FundAcc(gaugeCreator, coins)
-	distrTo := lockuptypes.QueryCondition{
-		LockQueryType: lockuptypes.ByDuration,
-		Denom:         "lptoken",
-		Duration:      time.Second,
-	}
 
-	// mints coins so supply exists on chain
-	mintLPtokens := sdk.Coins{sdk.NewInt64Coin(distrTo.Denom, 200)}
-	suite.FundAcc(gaugeCreator, mintLPtokens)
-
-	gaugeId, err := suite.App.IncentivesKeeper.CreateGauge(suite.Ctx, true, gaugeCreator, coins, distrTo, time.Now(), 1)
-	suite.NoError(err)
-	err = suite.App.PoolIncentivesKeeper.UpdateDistrRecords(suite.Ctx, poolincentivestypes.DistrRecord{
-		GaugeId: gaugeId,
-		Weight:  sdk.NewInt(100),
-	})
-	suite.NoError(err)
-
-	// At this time, there is no distr record, so the asset should be allocated to the community pool.
-	mintCoin := sdk.NewCoin("stake", sdk.NewInt(100000))
-	mintCoins := sdk.Coins{mintCoin}
-	err = mintKeeper.MintCoins(suite.Ctx, mintCoins)
-	suite.NoError(err)
-	err = mintKeeper.DistributeMintedCoin(suite.Ctx, mintCoin)
-	suite.NoError(err)
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			if tc.hasPreExistingSupply {
+				fee := sdk.NewCoin("stake", sdk.NewInt(100000))
+				fees := sdk.NewCoins(fee)
+				err := simapp.FundModuleAccount(suite.App.BankKeeper,
+					suite.Ctx,
+					authtypes.FeeCollectorName,
+					fees)
+				suite.NoError(err)
+			}
 
-	feePool := suite.App.DistrKeeper.GetFeePool(suite.Ctx)
-	feeCollector := suite.App.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName)
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.Staking).TruncateInt(),
-		suite.App.BankKeeper.GetAllBalances(suite.Ctx, feeCollector).AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.CommunityPool),
-		feePool.CommunityPool.AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.DeveloperRewards).TruncateInt(),
-		suite.App.BankKeeper.GetBalance(suite.Ctx, devRewardsReceiver, "stake").Amount)
+			coin := suite.App.MintKeeper.GetProportions(suite.Ctx, tc.fee, tc.ratio)
+			suite.Equal(tc.expectedCoin, coin)
+		})
+	}
+}
 
-	// Test for multiple dev reward addresses
-	params.WeightedDeveloperRewardsReceivers = []types.WeightedAddress{
+func (suite *KeeperTestSuite) TestDistrAssetToDeveloperRewardsAddr() {
+	var (
+		distrTo = lockuptypes.QueryCondition{
+			LockQueryType: lockuptypes.ByDuration,
+			Denom:         "lptoken",
+			Duration:      time.Second,
+		}
+		params       = suite.App.MintKeeper.GetParams(suite.Ctx)
+		gaugeCoins   = sdk.Coins{sdk.NewInt64Coin("stake", 10000)}
+		gaugeCreator = sdk.AccAddress([]byte("addr2---------------"))
+		mintLPtokens = sdk.Coins{sdk.NewInt64Coin(distrTo.Denom, 200)}
+	)
+
+	tests := []struct {
+		name              string
+		weightedAddresses []types.WeightedAddress
```

Comment:
Can we add test cases with nil weightedAddresses and then test if distribution makes it go to the community pool?

Reply:
I think we should wait for #1916. 
This issue will change [DistributeMintedCoin](https://github.com/osmosis-labs/osmosis/blob/97e1bc60a441a5fbeaff5ff94c6f241973b3da28/x/mint/keeper/keeper.go#L194) logic.

Reply:
The refactoring in #1916 will be focused on clean up and further testability only. I tend to agree with @mattverse that such a test could be useful

Reply:
I added this test.